src/main/java/ay2021s1_cs2103_w16_3/finesse/model/transaction/Amount.java

```diff
@@ -26,6 +26,7 @@ public Amount(String amount) {
         requireNonNull(amount);
         checkArgument(isValidAmount(amount), MESSAGE_CONSTRAINTS);
         value = new BigDecimal(amount.replaceFirst("^\\$", ""));
+        assert value.compareTo(BigDecimal.ZERO) >= 0; // amount should be non-negative
```

Comment:
Create an issue for this lest we forget to update the actual input validation rules?